packages/account/src/Sections/Profile/PersonalDetails/__tests__/personal-details.spec.tsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import { cleanup, render, waitFor, screen, fireEvent } from '@testing-library/react';
+import { createBrowserHistory } from 'history';
+import { Router } from 'react-router';
+import { PersonalDetailsForm } from '../personal-details';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import userEvent from '@testing-library/user-event';
+
+afterAll(cleanup);
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Loading: () => <div>Loading</div>,
+}));
+
+jest.mock('@deriv/shared/src/services/ws-methods', () => ({
+    __esModule: true, // this property makes it work,
+    default: 'mockedDefaultExport',
+    WS: {
+        wait: (...payload: []) => {
+            return Promise.resolve([...payload]);
+        },
+    },
+    useWS: () => undefined,
+}));
+
+describe('<PersonalDetailsForm />', () => {
+    const history = createBrowserHistory();
+
+    const promise = Promise.resolve();
```

Comment:
this Promise is not resolved with any value, do we really need it? I mean, why just not to use jest.fn()
and also, could you please check mockStore, some of the fields are defined and we don’t need to specify them here again

Reply:
We have to specify the promise here since fetchResidenceList and fetchStatesList will return a promise value. If not the test case will fail stating that the both function that I tried to call, `.then` is not a function

Reply:
alright, thanks for reply.
I was curious because here we not return any value :)